src/core/meshcop/tcat_agent.cpp

```diff
@@ -540,6 +547,95 @@ Error TcatAgent::HandlePing(const Message &aIncomingMessage,
     return error;
 }
 
+Error TcatAgent::HandleGetNetworkName(Message &aOutgoingMessage)
+{
+    Error       error  = kErrorNone;
+    const char *name   = otThreadGetNetworkName(&GetInstance());
+    uint16_t    length = static_cast<uint16_t>(strlen(name));
+    ot::Tlv     tlv;
+
+    VerifyOrExit(Get<ActiveDatasetManager>().IsCommissioned(), error = kErrorInvalidState);
+    VerifyOrExit(length > 0, error = kErrorInvalidState);
+
+    tlv.SetType(kTlvResponseWithPayload);
+    tlv.SetLength(static_cast<uint8_t>(length));
+    SuccessOrExit(error = aOutgoingMessage.Append(tlv));
+    SuccessOrExit(error = aOutgoingMessage.AppendBytes(name, length));
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetDeviceId(Message &aOutgoingMessage)
+{
+    uint8_t deviceId[8];
+    uint8_t length = 0;
+    ot::Tlv tlv;
+    Error   error = kErrorNone;
+
+    if (mVendorInfo->mDeviceIds != nullptr)
+    {
+        // If array is not null return first set device ID.
+        for (uint8_t i = 0; mVendorInfo->mDeviceIds[i].mDeviceIdType != OT_TCAT_DEVICE_ID_EMPTY; i++)
+        {
+            length = mVendorInfo->mDeviceIds->mDeviceIdLen;
+            memcpy(deviceId, mVendorInfo->mDeviceIds[i].mDeviceId, length);
+            break;
+        }
+    }
+
+    if (length == 0)
+    {
+        // If array is null or contains only empty entries return EUI64.
+        otPlatRadioGetIeeeEui64(&GetInstance(), deviceId);
+        length = 8;
+    }
+
+    tlv.SetType(kTlvResponseWithPayload);
+    tlv.SetLength(static_cast<uint8_t>(length));
```

Comment:
isn't `length` already of `uint8_t` type?

Reply:
I guess it could be rewritten to be `uint16_t` to keep the code similar to functions implemented below?